deflect/ImageSegmenter.cpp

```diff
@@ -163,113 +173,92 @@ bool ImageSegmenter::_generateRaw(const ImageWrapper& image,
     return true;
 }
 
-void ImageSegmenter::setNominalSegmentDimensions(const unsigned int width,
-                                                 const unsigned int height)
+Segments ImageSegmenter::_generateSegments(const ImageWrapper& image) const
 {
-    _nominalSegmentWidth = width;
-    _nominalSegmentHeight = height;
-}
+    Segments segments;
+    for (const auto& params : _makeSegmentParameters(image))
+    {
+        Segment segment;
+        segment.parameters = params;
+        segment.view =
+            image.view == View::side_by_side ? View::left_eye : image.view;
+        segment.sourceImage = &image;
+        segments.push_back(segment);
+    }
 
-#ifdef UNIORM_SEGMENT_WIDTH
-SegmentParametersList ImageSegmenter::generateSegmentParameters(
-    const ImageWrapper& image) const
-{
-    unsigned int numSubdivisionsX = 1;
-    unsigned int numSubdivisionsY = 1;
+    if (image.view == View::side_by_side)
+    {
+        if (image.width % 2 != 0)
+            throw std::runtime_error("side_by_side image width must be even!");
 
-    unsigned int uniformSegmentWidth = image.width;
-    unsigned int uniformSegmentHeight = image.height;
+        // create copy of segments for right view
+        auto segmentsRight = segments;
+        for (auto&& segment : segmentsRight)
```

Comment:
Yeah, in this case auto&& it's not an rvalue reference but a universal reference... I amended it to avoid confusion. Using auto& or auto&& make no difference here, but I let you google about it.